software/controller/test/commfail_alarm/commfail_alarm_test.cpp

```diff
@@ -0,0 +1,42 @@
+/* Copyright 2020, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "commfail_alarm.h"
+#include "gtest/gtest.h"
+#include <cmath>
+#include <tuple>
+
+TEST(CommFailAlarmTest, RaiseClearAlarm) {
```

Comment:
I don't understand what this is testing. I do not think this is a test. This is just something that runs the code.

Reply:
I will check

Reply:
added test cases for
Handler function
CommunicationFailed function
CommunicationResumed function

Reply:
Those are not test cases. Those are functions. For each function you must have multiple test cases.